diff --git a/KernelSU-Next/kernel/infra/file_wrapper.c b/KernelSU-Next/kernel/infra/file_wrapper.c
index dbad19bc..b8656eec 100644
--- a/KernelSU-Next/kernel/infra/file_wrapper.c
+++ b/KernelSU-Next/kernel/infra/file_wrapper.c
@@ -12,7 +12,10 @@
 #include <linux/uaccess.h>
 #include <linux/version.h>
 #include <linux/mount.h>
-
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4, 16, 0)
+typedef unsigned int __poll_t;
+#endif
+ 
 #include "objsec.h"
 
 #include "klog.h" // IWYU pragma: keep
@@ -99,7 +102,7 @@ static int ksu_wrapper_iopoll(struct kiocb *kiocb, struct io_comp_batch *icb,
     kiocb->ki_filp = orig;
     return orig->f_op->iopoll(kiocb, icb, v);
 }
-#else
+#elif LINUX_VERSION_CODE >= KERNEL_VERSION(5, 1, 0)
 static int ksu_wrapper_iopoll(struct kiocb *kiocb, bool spin)
 {
     struct ksu_file_wrapper *data = kiocb->ki_filp->private_data;
@@ -330,6 +330,7 @@ static ssize_t ksu_wrapper_copy_file_range(struct file *file_in, loff_t pos_in,
 // https://cs.android.com/android/kernel/superproject/+/common-android-mainline:common/fs/remap_range.c;l=403-404;drc=398da7defe218d3e51b0f3bdff75147e28125b60
 // REMAP_FILE_DEDUP: use file_out
 // https://cs.android.com/android/kernel/superproject/+/common-android-mainline:common/fs/remap_range.c;l=483-484;drc=398da7defe218d3e51b0f3bdff75147e28125b60
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 6, 0)
 static loff_t ksu_wrapper_remap_file_range(struct file *file_in, loff_t pos_in,
                                            struct file *file_out,
                                            loff_t pos_out, loff_t len,
@@ -358,6 +359,7 @@ static int ksu_wrapper_fadvise(struct file *fp, loff_t off1, loff_t off2,
     }
     return -EINVAL;
 }
+#endif

 static void ksu_release_file_wrapper(struct ksu_file_wrapper *data);

@@ -389,7 +391,9 @@ static struct ksu_file_wrapper *ksu_create_file_wrapper(struct file *fp)
     p->ops.write = fp->f_op->write ? ksu_wrapper_write : NULL;
     p->ops.read_iter = fp->f_op->read_iter ? ksu_wrapper_read_iter : NULL;
     p->ops.write_iter = fp->f_op->write_iter ? ksu_wrapper_write_iter : NULL;
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 1, 0)
     p->ops.iopoll = fp->f_op->iopoll ? ksu_wrapper_iopoll : NULL;
+#endif
 #if LINUX_VERSION_CODE < KERNEL_VERSION(6, 6, 0)
     p->ops.iterate = fp->f_op->iterate ? ksu_wrapper_iterate : NULL;
 #endif
@@ -403,7 +407,7 @@ static struct ksu_file_wrapper *ksu_create_file_wrapper(struct file *fp)
     p->ops.mmap = fp->f_op->mmap ? ksu_wrapper_mmap : NULL;
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 12, 0)
     p->ops.fop_flags = fp->f_op->fop_flags;
-#else
+#elif LINUX_VERSION_CODE >= KERNEL_VERSION(4, 20, 0)
     p->ops.mmap_supported_flags = fp->f_op->mmap_supported_flags;
 #endif
     p->ops.flush = fp->f_op->flush ? ksu_wrapper_flush : NULL;
@@ -426,9 +430,13 @@ static struct ksu_file_wrapper *ksu_create_file_wrapper(struct file *fp)
     p->ops.show_fdinfo = fp->f_op->show_fdinfo ? ksu_wrapper_show_fdinfo : NULL;
     p->ops.copy_file_range =
         fp->f_op->copy_file_range ? ksu_wrapper_copy_file_range : NULL;
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 6, 0)
     p->ops.remap_file_range =
         fp->f_op->remap_file_range ? ksu_wrapper_remap_file_range : NULL;
+#endif
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 6, 0)
     p->ops.fadvise = fp->f_op->fadvise ? ksu_wrapper_fadvise : NULL;
+#endif

 #if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 6, 0)
     p->ops.splice_eof = fp->f_op->splice_eof ? ksu_wrapper_splice_eof : NULL;
@@ -489,11 +497,13 @@ ksu_anon_inode_make_secure_inode(const char *name,
     if (IS_ERR(inode))
         return inode;
     inode->i_flags &= ~S_PRIVATE;
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 4, 0)
     error = security_inode_init_security_anon(inode, &qname, context_inode);
     if (error) {
         iput(inode);
         return ERR_PTR(error);
     }
+#endif
     return inode;
 }

@@ -513,8 +523,24 @@ static struct file *ksu_anon_inode_create_getfile_compat(
         goto err;
     }

+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 1, 0)
     file = alloc_file_pseudo(inode, anon_inode_mnt, name,
                              flags & (O_ACCMODE | O_NONBLOCK), fops);
+#else
+    {
+        struct path path = {
+            .mnt = anon_inode_mnt,
+            .dentry = d_alloc_name(anon_inode_mnt->mnt_root, name),
+        };
+        if (!path.dentry) {
+            file = ERR_PTR(-ENOMEM);
+        } else {
+            file = alloc_file(&path, flags & (O_ACCMODE | O_NONBLOCK), fops);
+            if (IS_ERR(file))
+                dput(path.dentry);
+        }
+    }
+#endif
     if (IS_ERR(file))
         goto err_iput;

@@ -568,7 +594,12 @@ int ksu_install_file_wrapper(int fd)
     struct inode *wrapper_inode = file_inode(wrapper_file);
     // libc's stdio relies on the fstat() result of the fd to determine its buffer type.
     wrapper_inode->i_mode = file_inode(orig_file)->i_mode;
-    struct inode_security_struct *wrapper_sec = selinux_inode(wrapper_inode);
+    struct inode_security_struct *wrapper_sec;
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 2, 0)
+    wrapper_sec = selinux_inode(wrapper_inode);
+#else
+    wrapper_sec = wrapper_inode->i_security;
+#endif
     // Use ksu_file_sid to bypass SELinux check.
     // When we call `su` from terminal app, this is useful.
     if (wrapper_sec) {
