--- a/KernelSU-Next/kernel/hook/syscall_hook.h
+++ b/KernelSU-Next/kernel/hook/syscall_hook.h
@@ -1,8 +1,11 @@
 #ifndef __KSU_H_KSU_SYSCALL_HOOK
 #define __KSU_H_KSU_SYSCALL_HOOK
-#include <asm/syscall.h>
 
 #if defined(__x86_64__)
+#include <asm/syscall.h>
 typedef sys_call_ptr_t syscall_fn_t;
+#elif defined(__aarch64__)
+#include <linux/ptrace.h>
+typedef long (*syscall_fn_t)(const struct pt_regs *);
 #endif
