diff --git a/KernelSU-Next/kernel/infra/su_mount_ns.c b/KernelSU-Next/kernel/infra/su_mount_ns.c
--- a/KernelSU-Next/kernel/infra/su_mount_ns.c
+++ b/KernelSU-Next/kernel/infra/su_mount_ns.c
@@ -28,27 +28,31 @@
                       const char *type_page, unsigned long flags,
                       void *data_page);
 
-#if defined(__aarch64__)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 0, 0) && defined(__aarch64__)
 extern long __arm64_sys_setns(const struct pt_regs *regs);
-#elif defined(__x86_64__)
+#elif LINUX_VERSION_CODE >= KERNEL_VERSION(5, 0, 0) && defined(__x86_64__)
 extern long __x64_sys_setns(const struct pt_regs *regs);
 #endif
 
 static long ksu_sys_setns(int fd, int flags)
 {
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 0, 0)
     struct pt_regs regs;
     memset(&regs, 0, sizeof(regs));
 
     PT_REGS_PARM1(&regs) = fd;
     PT_REGS_PARM2(&regs) = flags;
 
-#if defined(__aarch64__)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 0, 0) && defined(__aarch64__)
     return __arm64_sys_setns(&regs);
-#elif defined(__x86_64__)
+#elif LINUX_VERSION_CODE >= KERNEL_VERSION(5, 0, 0) && defined(__x86_64__)
     return __x64_sys_setns(&regs);
 #else
 #error "Unsupported arch"
 #endif
+#else
+    return sys_setns(fd, flags);
+#endif
 }
 
 // global mode , need CAP_SYS_ADMIN and CAP_SYS_CHROOT to perform setns
@@ -158,7 +162,11 @@
     // make root mount private
     struct path root_path;
     get_fs_root(current->fs, &root_path);
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 0, 0)
     int pm_ret = path_mount(NULL, &root_path, NULL, MS_PRIVATE | MS_REC, NULL);
+#else
+    int pm_ret = 0;
+#endif
     path_put(&root_path);
 
     if (pm_ret < 0) {
