diff --git a/KernelSU-Next/kernel/include/klog.h b/KernelSU-Next/kernel/include/klog.h
index a934027f..78e4cb6d 100644
--- a/KernelSU-Next/kernel/include/klog.h
+++ b/KernelSU-Next/kernel/include/klog.h
@@ -2,10 +2,19 @@
 #define __KSU_H_KLOG
 
 #include <linux/printk.h>
+#include <linux/version.h>
 
 #ifdef pr_fmt
 #undef pr_fmt
 #define pr_fmt(fmt) "KernelSU: " fmt
 #endif
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4, 15, 0)
+#define __poll_t unsigned int
+#endif
+
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4, 20, 0)
+#define REMAP_FILE_DEDUP (1 << 0)
+#endif
+
 #endif
