--- a/KernelSU-Next/kernel/include/util.h
+++ b/KernelSU-Next/kernel/include/util.h
@@ -2,11 +2,20 @@
 #define __KSU_H_UTIL
 
 #include "linux/fdtable.h" // IWYU pragma: keep
+#include <linux/uaccess.h>
 #include <linux/version.h>
 #include <linux/syscalls.h>
 
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 11, 0)
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 2, 0)
+#define ksu_close_fd(fd) sys_close(fd)
+#elif LINUX_VERSION_CODE < KERNEL_VERSION(5, 11, 0)
+#define ksu_close_fd ksys_close
+#else
 #define ksu_close_fd close_fd
-#else
-#define ksu_close_fd ksys_close
 #endif
+
+#ifndef strncpy_from_user_nofault
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 3, 0)
+#define strncpy_from_user_nofault(dst, src, n) strncpy_from_user(dst, (const char __user *)(src), n)
+#endif
+#endif
