diff --git a/KernelSU-Next/kernel/hook/arm64/patch_memory.c b/KernelSU-Next/kernel/hook/arm64/patch_memory.c
index 0c2576da..5aa3b371 100644
--- a/KernelSU-Next/kernel/hook/arm64/patch_memory.c
+++ b/KernelSU-Next/kernel/hook/arm64/patch_memory.c
@@ -12,6 +12,19 @@
 #include "linux/stop_machine.h"
 #include "asm/cacheflush.h"
 #include "asm-generic/fixmap.h"
+#include <linux/version.h>
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 5, 0)
+#define __p4d_to_phys(p4d)  (p4d_val(p4d) & PHYS_MASK)
+#define __pud_to_phys(pud)  (pud_val(pud) & PHYS_MASK)
+#define __pmd_to_phys(pmd)  (pmd_val(pmd) & PHYS_MASK)
+#define __pte_to_phys(pte)  (pte_val(pte) & PHYS_MASK)
+#endif
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 4, 0)
+#define copy_to_kernel_nofault(dst, src, len) probe_kernel_write(dst, src, len)
+#endif
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 0, 0)
+#define __flush_icache_range flush_icache_range
+#endif

 // https://github.com/fuqiuluo/ovo/blob/f7da411458e87d32438dc14fce5a3313ed0c967e/ovo/mmuhack.c#L21
