--- a/KernelSU-Next/kernel/include/util.h
+++ b/KernelSU-Next/kernel/include/util.h
@@ -2,11 +2,22 @@
 #define __KSU_H_UTIL
 
 #include "linux/fdtable.h" // IWYU pragma: keep
+#include <linux/uaccess.h>
 #include <linux/version.h>
 #include <linux/syscalls.h>
 
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 11, 0)
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 2, 0)
+#define ksu_close_fd(fd) sys_close(fd)
+#elif LINUX_VERSION_CODE < KERNEL_VERSION(5, 11, 0)
+#define ksu_close_fd ksys_close
+#else
 #define ksu_close_fd close_fd
-#else
-#define ksu_close_fd ksys_close
 #endif
+
+#ifndef strncpy_from_user_nofault
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 3, 0)
+#define strncpy_from_user_nofault(dst, src, n) strncpy_from_user(dst, (const char __user *)(src), n)
+#endif
+#endif

--- a/KernelSU-Next/kernel/sulog/event.c
+++ b/KernelSU-Next/kernel/sulog/event.c
@@ -12,6 +12,8 @@
 #include <linux/mm.h>
 #endif
 
+#include "util.h"
+
 #include "feature/sulog.h"
 #include "infra/event_queue.h"
 #include "klog.h" // IWYU pragma: keep

--- a/KernelSU-Next/kernel/runtime/ksud_integration.c
+++ b/KernelSU-Next/kernel/runtime/ksud_integration.c
@@ -23,6 +23,8 @@
 #include <linux/stat.h>
 #include <linux/uio.h>
 
+#include "util.h"
+
 #include "arch.h"
 #include "klog.h" // IWYU pragma: keep
 #include "ksu.h"
--- a/KernelSU-Next/kernel/feature/sucompat.c
+++ b/KernelSU-Next/kernel/feature/sucompat.c
@@ -100,7 +100,7 @@ long ksu_handle_faccessat_sucompat(int orig_nr, struct pt_regs *regs)
 	char path[sizeof(su_path) + 1];
 	memset(path, 0, sizeof(path));
-	strncpy_from_user_nofault(path, *filename_user, sizeof(path));
+	strncpy_from_user(path, *filename_user, sizeof(path));
 
 	if (unlikely(!memcmp(path, su_path, sizeof(su_path)))) {
 		old_cred = override_creds(ksu_cred);
@@ -136,7 +136,7 @@ long ksu_handle_stat_sucompat(int orig_nr, struct pt_regs *regs)
 	char path[sizeof(su_path) + 1];
 	memset(path, 0, sizeof(path));
-	strncpy_from_user_nofault(path, *filename_user, sizeof(path));
+	strncpy_from_user(path, *filename_user, sizeof(path));
 
 	if (unlikely(!memcmp(path, su_path, sizeof(su_path)))) {
 		old_cred = override_creds(ksu_cred);
