diff --git a/KernelSU-Next/kernel/infra/seccomp_cache.c b/KernelSU-Next/kernel/infra/seccomp_cache.c
index 6511066b..bb49c14e 100644
--- a/KernelSU-Next/kernel/infra/seccomp_cache.c
+++ b/KernelSU-Next/kernel/infra/seccomp_cache.c
@@ -5,6 +5,12 @@
 #include <linux/uaccess.h>
 #include <linux/filter.h>
 #include <linux/seccomp.h>
+#include <asm/unistd.h>
+
+#ifndef SECCOMP_ARCH_NATIVE_NR
+#define SECCOMP_ARCH_NATIVE_NR __NR_syscalls
+#endif
+
 #include "klog.h" // IWYU pragma: keep
 #include "infra/seccomp_cache.h"
