diff --git a/KernelSU-Next/kernel/supercall/supercall.c b/KernelSU-Next/kernel/supercall/supercall.c
index 53e9a0eb..2af4e868 100644
--- a/KernelSU-Next/kernel/supercall/supercall.c
+++ b/KernelSU-Next/kernel/supercall/supercall.c
@@ -11,6 +11,9 @@
 #include <linux/syscalls.h>
 #include <linux/task_work.h>
 #include <linux/uaccess.h>
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 11, 0)
+#define TWA_RESUME true
+#endif
 #include <linux/version.h>
 #include <linux/utsname.h> // utsname() and uts_sem
 
