diff --git a/KernelSU-Next/kernel/hook/syscall_hook.h b/KernelSU-Next/kernel/hook/syscall_hook.h
index 944f43f6..c1f40264 100644
--- a/KernelSU-Next/kernel/hook/syscall_hook.h
+++ b/KernelSU-Next/kernel/hook/syscall_hook.h
@@ -2,7 +2,9 @@
 #define __KSU_H_KSU_SYSCALL_HOOK
 #include <asm/syscall.h>

-#if defined(__x86_64__)
+#if defined(__aarch64__)
+typedef long (*syscall_fn_t)(const struct pt_regs *);
+#elif defined(__x86_64__)
 typedef sys_call_ptr_t syscall_fn_t;
 #endif
