--- a/KernelSU-Next/kernel/feature/sucompat.c
+++ b/KernelSU-Next/kernel/feature/sucompat.c
@@ -97,7 +97,7 @@ long ksu_handle_faccessat_sucompat(int orig_nr, struct pt_regs *regs)
 	char path[sizeof(su_path) + 1];
 	memset(path, 0, sizeof(path));
-	strncpy_from_user_nofault(path, *filename_user, sizeof(path));
+	strncpy_from_user(path, *filename_user, sizeof(path));
 
 	if (unlikely(!memcmp(path, su_path, sizeof(su_path)))) {
 		old_cred = override_creds(ksu_cred);
@@ -133,7 +133,7 @@ long ksu_handle_stat_sucompat(int orig_nr, struct pt_regs *regs)
 	char path[sizeof(su_path) + 1];
 	memset(path, 0, sizeof(path));
-	strncpy_from_user_nofault(path, *filename_user, sizeof(path));
+	strncpy_from_user(path, *filename_user, sizeof(path));
 
 	if (unlikely(!memcmp(path, su_path, sizeof(su_path)))) {
 		old_cred = override_creds(ksu_cred);
